README.md

```diff
@@ -1,2 +1,66 @@
-NServiceBus.Testing
-===================
+## A testing framework for NServiceBus
+
+This library contains a set of testable implementations (test-doubles) for various NServiceBus components, making testing things like handlers, sagas and behaviors easy.
+
+### Getting started
+
+To get started, install the package via NuGet:
+
+```
+Install-Package NServiceBus.Testing
+``` 
+
+### Example
+
+Given a handler:
+
+```csharp
+public class MyHandler : IHandleMessages<RequestMessage>
+{
+    public async Task Handle(RequestMessage message, IMessageHandlerContext context)
+    {
+        ResponseMessage reply = new ResponseMessage
+        {
+            String = message.String
+        };
+        await context.Reply(reply);
+    }
+}
+```
+
+A test with the NServiceBus.Testing framework looks like this:
+
+
+```csharp
+[TestFixture]
+public class Tests
+{
+    [Test]
+    public void Run()
+    {
+        Test.Handler<MyHandler>()
+            .ExpectReply<ResponseMessage>(m => m.String == "hello")
+            .OnMessage<RequestMessage>(m => m.String = "hello");
+    }
+}
+```
+
+This test verifies that the handler replies with the same message text that was in the request.
+
+### Arrange-Act-Assert (AAA) style tests
+
+If you prefer not using the Fluent Test API for composing your tests, you can use the provided testable implementations to assert against:
+
+```csharp
+[Test]
+public async Task ShouldReplyWithResponseMessage()
+{
+    var handler = new MyHandler();
+    var fakeContext = new TestableMessageHandlerContext();
+
+    await handler.Handle(new RequestMessage(), fakeContext);
+
+    Assert.AreEqual(1, fakeContext.RepliedMessages.Length);
+    Assert.IsInstanceOf<ResponseMessage>(fakeContext.RepliedMessages[0].Message);
```

Comment:
@timbussmann maybe, but that relies even more on the extension methods. This one shows the real bare minimum (both tests make it obvious `RepliedMessages` is an array/list.